bagel/utilities/derivative_utils.py

```diff
@@ -17,35 +18,92 @@
+        f"Allowed pipeline names are the following pipelines supported natively in Nipoppy (https://github.com/nipoppy/pipeline-catalog):\n"
+        f"{list(mappings.KNOWN_PIPELINE_URIS.keys())}"
+    )
+    recognized_pipelines = list(
+        set(pipelines).intersection(mappings.KNOWN_PIPELINE_URIS)
+    )
     unrecognized_pipelines = list(
         set(pipelines).difference(mappings.KNOWN_PIPELINE_URIS)
     )
-    if len(unrecognized_pipelines) > 0:
+
+    if not recognized_pipelines:
         raise LookupError(
+            f"The processing status file contains no recognized pipelines in the column '{PROC_STATUS_COLS['pipeline_name']}'.\n"
+            f"{allowed_pipelines_message}"
+        )
+    if unrecognized_pipelines:
+        warnings.warn(
             f"The processing status file contains unrecognized pipelines in the column '{PROC_STATUS_COLS['pipeline_name']}': "
-            f"{unrecognized_pipelines}. "
-            f"Allowed pipeline names are the following pipelines supported natively in Nipoppy (https://github.com/nipoppy/pipeline-catalog): \n"
-            f"{mappings.KNOWN_PIPELINE_URIS}"
+            f"{unrecognized_pipelines}. These will be ignored.\n"
+            f"{allowed_pipelines_message}"
         )
+    return recognized_pipelines
 
 
-def check_pipeline_versions_are_recognized(
+def classify_pipeline_versions(
     pipeline: str, versions: Iterable[str]
-):
+) -> tuple[list, list]:
     """
-    Check that all pipeline versions in the processing status file are supported by Nipoppy.
-    Assumes that the input pipeline name is recognized.
+    For a given pipeline, return the recognized and unrecognized pipeline versions in the processing status file
+    based on the Nipoppy pipeline catalog, and return both as lists.
     """
+    recognized_versions = list(
+        set(versions).intersection(mappings.KNOWN_PIPELINE_VERSIONS[pipeline])
+    )
     unrecognized_versions = list(
         set(versions).difference(mappings.KNOWN_PIPELINE_VERSIONS[pipeline])
     )
-    if len(unrecognized_versions) > 0:
+
+    return recognized_versions, unrecognized_versions
+
+
+def check_at_least_one_pipeline_version_is_recognized(status_df: pd.DataFrame):
+    """
+    Check that at least one pipeline name and version combination found in the processing status file is supported by Nipoppy.
+    """
+    more_info_message = (
+        "Allowed processing pipelines and versions are those supported natively in Nipoppy. "
+        "For a full list, see https://github.com/nipoppy/pipeline-catalog."
+    )
+
+    recognized_pipelines = get_recognized_pipelines(
+        status_df[PROC_STATUS_COLS["pipeline_name"]].unique()
+    )
+
+    any_recognized_versions = 0
+    unrecognized_pipeline_versions = {}
+    for pipeline in recognized_pipelines:
+        versions = status_df[
+            status_df[PROC_STATUS_COLS["pipeline_name"]] == pipeline
+        ][PROC_STATUS_COLS["pipeline_version"]].unique()
```

Comment:
Hmm, I'm not sure I agree - this is what it would look like with .query():

```
versions = status_df.query(f'{PROC_STATUS_COLS["pipeline_name"]} == @pipeline')[
    PROC_STATUS_COLS["pipeline_version"]
].unique()
```

to me, this syntax doesn't provide any obvious improvement in clarity 🤔

Reply:
ah, good point. I didn't notice that the pipeline_name came from a different DF and so we don't get around the `PROC_STATUS_COLS["pipeline_name"]` bit.